### ISSUE TITLE
error building mpich 3.2.1 on macOS

### ISSUE THREAD

**Issue by cyrush** [spack/spack#8432]
### Expected Result
spack to install  mpich 3.2.1 (default version)

### Actual Result

mpich 3.2.1 fails to build due to a rejected patch.

It appears the patch was added  for 3.2 and it is not needed for 3.2.1.

### Steps to reproduce the issue

```console
$ bin/spack install mpich%clang 
==> Installing mpich
==> Using cached archive: /Users/harrison37/Work/alpine/spack/var/spack/cache/mpich/mpich-3.2.1.tar.gz
==> Staging archive: /Users/harrison37/Work/alpine/spack/var/spack/stage/mpich-3.2.1-gmk64r7kkptknk4e2knl7zlof6tqirfo/mpich-3.2.1.tar.gz
==> Created stage in /Users/harrison37/Work/alpine/spack/var/spack/stage/mpich-3.2.1-gmk64r7kkptknk4e2knl7zlof6tqirfo
1 out of 1 hunk FAILED -- saving rejects to file src/include/mpiimpl.h.rej
==> Patch mpich32_clang.patch failed.
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/Users/harrison37/Work/alpine/spack/var/spack/repos/builtin/packages/mpich/mpich32_clang.patch' '-d' '.'
==> Error: [Errno 2] No such file or directory: '/Users/harrison37/Work/alpine/spack/var/spack/stage/mpich-3.2.1-gmk64r7kkptknk4e2knl7zlof6tqirfo/mpich-3.2.1/spack-build.out'...
```

Note: 3.2 does build:

```console
bin/spack install mpich@3.2%clang
==> Installing mpich
==> Fetching http://www.mpich.org/static/downloads/3.2/mpich-3.2.tar.gz
######################################################################## 100.0%
==> Staging archive: /Users/harrison37/Work/alpine/spack/var/spack/stage/mpich-3.2-gea6brxmwnhbxy5xsj6xzp454jbgguta/mpich-3.2.tar.gz
==> Created stage in /Users/harrison37/Work/alpine/spack/var/spack/stage/mpich-3.2-gea6brxmwnhbxy5xsj6xzp454jbgguta
==> Applied patch mpich32_clang.patch
==> Building mpich [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
```

Here is how the patch is specified in the package -- how can we restrict to just `3.2` and not `3.2.x`? 

```python
patch('mpich32_clang.patch', when='@3.2%clang')
```

### System Info:

macOS 10.13
XCode 9 (clang reports as version 9)

-----

**Comment by cyrush** [comment]
@adamjstewart thanks, I searched for mpich in issues, but not PRs -- so I missed that thread!

**Comment by balay** [comment]
I stumbled on to this issue today..

**Comment by goxberry** [comment]
Just ran into this today. @adamjstewart, is there any chance #8320 could be revisited for review? It seems like that's the next action needed to move forward.

**Comment by adamjstewart** [comment]
Closed by #9330.
